gbpservice/neutron/nsf/core/main.py

```diff
@@ -0,0 +1,705 @@
+        self.binding_key = kwargs.get(
+            'binding_key') if 'binding_key' in kwargs else None
+        self.id = kwargs.get('id')
+        self.key = kwargs.get('key')
+        self.data = kwargs.get('data') if 'data' in kwargs else None
+        self.handler = kwargs.get('handler') if 'handler' in kwargs else None
+        self.poll_event = None  # Not to be used by user
+        self.worker_attached = None  # Not to be used by user
+        self.last_run = None  # Not to be used by user
+        self.max_times = -1  # Not to be used by user
+
+    def identify(self):
+        return "(id=%s,key=%s)" % (self.id, self.key)
+
+
+class EventCache(object):
+
+    def __init__(self, sc):
+        self._sc = sc
+        self._cache = core_queue.Queue(sc)
+
+    def rem(self, ev):
+        self._cache.remove([ev])
+
+    def rem_multi(self, evs):
+        self._cache.remove(evs)
+
+    def add(self, ev):
+        self._cache.put(ev)
+
+    def copy(self):
+        return self._cache.copy()
+
+
+class Serializer(object):
+
+    def __init__(self, sc):
+        self._sc = sc
+        '''
+        {'pid':{'binding_key':{'in_use':True, 'queue':[]}}}
+        '''
+        self._serializer_map = {}
+
+    def serialize(self, ev):
+        queued = False
+        LOG.debug(_("Serialize event %s" % (ev.identify())))
+        self._sc.lock()
+        if ev.worker_attached not in self._serializer_map:
+            self._serializer_map[ev.worker_attached] = {}
+        mapp = self._serializer_map[ev.worker_attached]
+        if ev.binding_key in mapp.keys():
+            queued = True
+            LOG.debug(_(
+                "There is already an event in progress"
+                "Queueing event %s" % (ev.identify())))
+
+            mapp[ev.binding_key]['queue'].append(ev)
+        else:
+            LOG.debug(_(
+                "Scheduling first event to exec"
+                "Event %s" % (ev.identify())))
+            mapp[ev.binding_key] = {'in_use': True, 'queue': []}
+        self._sc.unlock()
+        return queued
+
+    def deserialize(self, ev):
+        LOG.debug(_(
+            "Deserialize event %s"
+            % (ev.identify())))
+        self._sc.lock()
+        mapp = self._serializer_map[ev.worker_attached][ev.binding_key]
+        if mapp['queue'] == []:
+            LOG.debug(_(
+                "No more events in the serial Q -"
+                "Deleting the entry (%d) (%s)"
+                % (ev.worker_attached, ev.binding_key)))
+            del self._serializer_map[ev.worker_attached][ev.binding_key]
+        self._sc.unlock()
+
+    def copy(self):
+        self._sc.lock()
+        copy = dict(self._serializer_map)
+        self._sc.unlock()
+        return copy
+
+    def remove(self, ev):
+        self._sc.lock()
+        self._serializer_map[ev.worker_attached][
+            ev.binding_key]['queue'].remove(ev)
+        self._sc.unlock()
+
+
+class PollHandler(object):
```

Comment:
Comments in this class are suggestions for code readability. Use what you feel are appropriate.

PollQueue instead of PollHandler